src/unix/index_unix.ml

```diff
@@ -206,9 +206,9 @@ module IO : Index.IO = struct
     t.offset <- t.offset ++ len;
     if t.offset -- t.flushed > auto_flush_limit then sync t
 
-  let read t ~off buf =
+  let read t ~off ~len buf =
```

Comment:
What's the idea behind passing len to read instead of letting read determine the number of bytes it should read?

Reply:
We know in advance how many bytes we have to read, so It's to double-check that we have actually read the right amount of data.

Reply:
It's also no longer true that the number of bytes to be read is equal to the size of the buffer, since the buffer will always be 4096 bytes (and the number of bytes to be read is `<= 4096`).